tst/CTA.Rules.Test/RuleContributionTests.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using NUnit.Framework;
+
+namespace CTA.Rules.Test;
+
+/// <summary>
+/// This class is exclusively for testing RuleContributions.sln,
+/// a solution with a .NET Framework project containing all supported
+/// rules contributed by third parties.
+/// </summary>
+[TestFixture]
+class RuleContributionTests : AwsRulesBaseTest
+{
+    private string _tempDir = "";
+    private string _downloadLocation;
+    private Dictionary<string, TestSolutionAnalysis> _resultsDict;
+
+    [OneTimeSetUp]
+    public void Setup()
+    {
+        _tempDir = SetupTests.TempDir;
+        _downloadLocation = SetupTests.DownloadLocation;
+
+        var solutionName = "RuleContributions.sln";
+        var solutionPath = CopySolutionFolderToTemp(solutionName, _tempDir);
+        var net31Results = AnalyzeSolution(solutionPath, TargetFramework.DotnetCoreApp31);
+        var net50Results = AnalyzeSolution(solutionPath, TargetFramework.Dotnet5);
+        var net60Results = AnalyzeSolution(solutionPath, TargetFramework.Dotnet6);
+
+        _resultsDict = new Dictionary<string, TestSolutionAnalysis>
+        {
+            {TargetFramework.DotnetCoreApp31, net31Results},
+            {TargetFramework.Dotnet5, net50Results},
+            {TargetFramework.Dotnet6, net60Results}
+        };
+    }
+    
+    [TestCase(TargetFramework.Dotnet5)]
+    [TestCase(TargetFramework.Dotnet6)]
```

Comment:
what about 3.1?

Reply:
The rule that was submitted only included .NET 5 and .NET 6, so tests and implementation were of the same scope, though 3.1 was not attempted.

Reply:
since we know this rule is needed can we check compatibilty for 3,1 and add it now as well?

Reply:
Thanks, Sai. That's a good idea- will validate